### ISSUE TITLE
Megasync never finishes "Syncing..." even though no transfers are underway.

### ISSUE THREAD

**Issue by GH-A-CC** [meganz/MEGAsync#891]
I have one folder sync configured in MEGASync, this folder appears on the surface to be full synchronised but MEGASync never exits the "Syncing..." state.

**Operating System:** Fedora release 39 (Thirty Nine)
**MEGAsync version:**  4.11.0 (64 bit) (6d4c102)
**Internet connection:** >1Gbps both upload and download
**Sync'd folder details:** One folder ~1.4 TB in total, containing ~904,536 files in 29,368 subfolders
**Drive Details:** Drive the sync folder is on:  3.8TB btrfs formatted NVMe

I have what I believe to be a fully Synchronised folder, yet the MEGASync app remains in a "Syncing..." state and the icon never changes to the plain "M" logo without the miniature two curved arrows over it. 

I have twice tried using the "Settings -> Force a Full Scan" option where by the following behaviour occurs:

- The MEGASync app restarts.
- The app spends ~45+ minutes in "Scanning..." state, then transitions to "Syncing...".
- The app then sits in "Syncing..." state for what appears to be an indefinite amount of time - certainly in excess of 8 and 12 hours respectively in the two last times I have done it.

None of the files in the local or cloud based copy of the Sync'd folder are being changed.

In "Settings -> General" window the ".debris (local cache)", "SyncDebris (remote cache)", and "File versions" all show 0 bytes.

**Comment by GH-A-CC** [comment]
Comment to add this also happens on a different system set to synchronise the same folder - Fedora 38 this time, same folder sync settings and same MEGASync version.

**Comment by FGD-Garuda** [comment]
I'm facing the same issue, whether I use `megasync-bin` or `megasync-git` packages for Arch.

I think in my case it started when ICU package got updated to v74, some workarounds are required to run megasync until upstream fixes it.

Nevertheless, it always syncs non-stop and WRITES to my disk non-stop, even though nothing is synced. Constant disk I/O due to that.

**Comment by GH-A-CC** [comment]
> it always syncs non-stop and WRITES to my disk non-stop, even though nothing is synced. Constant disk I/O due to that.

So, I don't see this problem with non-stop writes. Your issue sounds like a different one - although potentially related. For me, `iotop` reports no reads or writes from MEGAsync after its switched over from "Scanning..." to "Syncing...".  I've also tested this by putting a sync on an obsolete non-SSD type drive mounted over USB, and I can observe the drive spins down and its power draw drops by 95%.

I have also recently tested if the file system format is relevant by rsync'ing the entire folder over to an NTFS formatted drive, deleting the sync pair in MEGAsync and re-creating it pointing to the new NTFS-backed folder copy.  I observe the same behaviour: i.e., MEGAsync spends about 45 mins "Scanning..." then switches to "Syncing...".

**Comment by FGD-Garuda** [comment]
Both issues started at the same time for me, there is a chance they aren't related, but I think there's a bigger chance they are.

I am under BTRFS filesystem and never experienced such constant I/Os from megasync for years. It's like every second, from 125kb to 500kb, most of the time in the 125-200kb range. I could dig down further and possibly find the files it's writing into but at the moment I don't see the advantage of investing such effort until someone who knows a lot better about these megasync issues ask me for troubleshooting data and knowing where he'd go with the result.

If it's a different issue I may open another bug for it, though. My point is I'd like that constant Syncing to be fixed and if it doesn't fix my constant I/Os then I open a bug specific for this.

**Comment by cihe13375** [comment]
I also encounter this issue sometimes for no obvious reasons (on both windows and linux), and usually I solve it in the following way:
1) log out of the current account (settings - account - log out)
2) quit megasync and launch again
3) login (type in username and password)
4) go to settings - sync and enable all syncs (they are automatically disabled upon logout)

Generally, I found this procedure solving many issues probably caused by syncs going into a bad internal state, which sometimes *cannot* be fixed by "force a full scan" (for whatever reason). idk if it would work for you though, as I had switched to 5.0.x for a long time and had forgot how 4.x version behaves...

btw you probably also want to report it using the "issue report" feature within the client, which likely gives a higher chance to get official responses

**Comment by FGD-Garuda** [comment]
Yes that does help! Logging out and back in actually fixed another issue I had.
Should have thought that but I guess it was too easy.

Tnx!

**Comment by GH-A-CC** [comment]
The proposed workaround from @cihe13375 that worked for @FGD-Garuda  has NOT worked for me.

**Comment by hugorn** [comment]
After updating, this count never ends and the synchronization remains in an infinite loop.
![image](https://github.com/meganz/MEGAsync/assets/9035153/9bcf3ed1-2077-467c-a789-1690d957bd75)

**Comment by Sascha-1** [comment]
The eternal syncing problem seems to be triggered when I delete a synced local file from my Linux Mint installation.

**Comment by Mamzu** [comment]
Hola, en mi caso, donde la carpeta que se sincronizaba era la de "Mis Documentos" se solucionó borrando las carpetas del sistema (accesos directos ocultos): "Mis Videos", "Mis Imagenes" y "Mi Musica" (que yo no usaba)
